pdp/static/js/pdp_raster_map.js

```diff
@@ -4,38 +4,72 @@
 
 "use strict";
 
-var CfTime = function (units, sDate) {
+var CfTime = function (units, sDate, calendar) {
     this.units = units;
     this.sDate = sDate;
+    this.calendar = calendar;
 };
 CfTime.prototype.setMaxTimeByIndex = function (index) {
     this.maxIndex =  index;
     this.eDate = this.toDate(index);
     return this.eDate;
 };
-CfTime.prototype.toDate = function (index) {
-    if (index === undefined) {
-        return this.sDate;
-    }
-    if (this.units === "days") {
-        var d = new Date(this.sDate.getTime());
-        d.setDate(this.sDate.getDate() + index);
-        return d;
-    }
+
+CfTime.prototype.toDate = function(index) {
+	if (index === undefined) {
+		return this.sDate;
+	}
+	var d = new Date(this.sDate.getTime());
+	if(this.units == "days") {
+		if(this.calendar != "standard") {
+			var daysPerYear = this.calendar == "365_day" ? 365 : 360;
+			d.setFullYear(this.sDate.getFullYear() + Math.floor(index / daysPerYear));
+	        var msPerDay = 1000 * 60 * 60 * 24;
+			var daysAlready = (d.getTime() - new Date(d.getFullYear(), 0, 1).getTime() ) / msPerDay;
+			var dayRemainder = (index % daysPerYear) + daysAlready;
+			if(dayRemainder >= daysPerYear) {
+				d.setFullYear(d.getFullYear() + 1);
+				dayRemainder = dayRemainder - daysPerYear;
+			}
+			if(this.calendar == "360_day") {
+				dayRemainder += Math.floor((dayRemainder / daysPerYear) * 5);
+			}
+			d.setTime(d.getTime() + dayRemainder);
+			
+		}
+		else{
+			d.setDate(this.sDate.getDate() + index);
+		}
+		return d;
+	}
 };
 CfTime.prototype.toIndex = function (d) {
     if (d < this.sDate || (this.eDate && this.eDate < d)) {
         return;
     }
-
-    if (this.units === "days") {
-        var msPerDay = 1000 * 60 * 60 * 24,
-            msDiff = d.getTime() - this.sDate.getTime(),
-            days = msDiff / msPerDay;
-        return Math.floor(days);
+    var days;
+    var msPerDay = 1000 * 60 * 60 * 24;
+    if(this.units == "days") {
+    	if(this.calendar != "standard") {
+    		var daysPerYear = this.calendar=="365_day" ? 365 : 360;
+    		days = (d.getFullYear() - this.sDate.getFullYear()) * daysPerYear;
+    		var steppedDate = new Date(d);
```

Comment:
It's the same date as the date-of-interest (the date we're looking for the index of) but "stepped backwards" a particular number of whole years. That is, if we're looking for the index of April 1 1970, and our start date is January 1 1950, the stepped-back date is April 1 1950. It's used to calculate the "remainder" days in the index that arise if the start date and the date-of-interest are different days of the year.

Since it's used to calculate ` remainderDays`, I renamed it `remainderDate`, hopefully that will be a little clearer.